corehq/apps/dump_reload/sql/filters.py

```diff
@@ -50,3 +50,59 @@ def get_filters(self, domain_name):
         for chunk in chunked(user_ids, 1000):
             query_kwarg = '{}__in'.format(self.user_id_field)
             yield Q(**{query_kwarg: chunk})
+
+
+class UnfilteredModelIteratorBuilder(object):
+    def __init__(self, model_label):
+        self.model_label = model_label
+        self.domain = self.model_class = self.db_alias = None
+
+    def prepare(self, domain, model_class, db_alias):
+        self.domain = domain
+        self.model_class = model_class
+        self.db_alias = db_alias
+        return self
+
+    def _base_queryset(self):
+        assert self.domain and self.model_class and self.db_alias, "Unprepared IteratorBuilder"
+        objects = self.model_class._default_manager
+        return objects.using(self.db_alias).order_by(self.model_class._meta.pk.name)
+
+    def querysets(self):
+        return self._base_queryset()
+
+    def iterators(self):
+        for queryset in self.querysets():
+            yield queryset.iterator()
+
+    def build(self, domain, model_class, db_alias):
+        return self.__class__(self.model_label).prepare(domain, model_class, db_alias)
+
+
+class FilteredModelIteratorBuilder(UnfilteredModelIteratorBuilder):
+    def __init__(self, model_label, filter):
+        super(FilteredModelIteratorBuilder, self).__init__(model_label)
+        self.filter = filter
+
+    def build(self, domain, model_class, db_alias):
+        return self.__class__(self.model_label, self.filter).prepare(domain, model_class, db_alias)
+
+    def querysets(self):
+        queryset = self._base_queryset()
+        filters = self.filter.get_filters(self.domain)
+        for filter in filters:
+            yield queryset.filter(filter)
+
+
+class UniqueFilteredModelIteratorBuilder(FilteredModelIteratorBuilder):
+    def iterators(self):
+        def _unique(iterator):
+            seen = set()
```

Comment:
will this accumulate the primary key of every row in the database? if yes, seems like that might take a lot of memory.